### ISSUE TITLE
复现#### Stage 2: Interleaved CoT SFT.出现问题

### ISSUE THREAD

**Issue by vansuu** [xinyan-cxy/MINT-CoT#3]
{'loss': 0.7716, 'grad_norm': 6.769598484039307, 'learning_rate': 9.861932938856016e-08, 'epoch': 0.06}                                                                                                                                 
  1%|██▎                                                                                                                                                                                           | 60/5064 [08:46<11:35:13,  8.34s/it]NaN or Inf found in input tensor.
{'loss': 6.8315, 'grad_norm': nan, 'learning_rate': 1.1834319526627217e-07, 'epoch': 0.07}                                                                                                                                              
  1%|██▋                                                                                                                                                                                           | 70/5064 [10:16<12:21:28,  8.91s/it]NaN or Inf found in input tensor.
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 1.3806706114398422e-07, 'epoch': 0.08}                                                                                                                                                 
  2%|███                                                                                                                                                                                           | 80/5064 [11:47<12:23:45,  8.95s/it]NaN or Inf found in input tensor.
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 1.5779092702169626e-07, 'epoch': 0.09}                                                                                                                                                 
  2%|███▏                                                                                                                                                                                          | 86/5064 [12:44<13:21:40,  9.66s/it]
参数没有改动，请问是哪里有问题呀

**Comment by vansuu** [comment]
> 感谢您报告此问题。我们已验证代码在我们这边正常运行。但是，由于硬件或软件环境的差异，可能会出现细微的差异。我们的实验基于**CUDA 12.1、NVIDIA A100（8 GPU）**和**PyTorch 2.5.1**进行。 为了解决此问题，我们建议尝试降低学习率或调整其他超参数以稳定训练。如果问题仍然存在，请告知我们，我们将很乐意为您提供进一步的帮助。

十分感谢您的回复，上述问题已解决。请问作者能否分享一下视觉交错CoT数据制作的脚本，这是否适用于不同类型的多模态数学推理数据，供社区调研～～～

**Comment by xinyan-cxy** [comment]
> > 感谢您报告此问题。我们已验证代码在我们这边正常运行。但是，由于硬件或软件环境的差异，可能会出现细微的差异。我们的实验基于**CUDA 12.1、NVIDIA A100（8 GPU）和PyTorch 2.5.1**进行。 为了解决此问题，我们建议尝试降低学习率或调整其他超参数以稳定训练。如果问题仍然存在，请告知我们，我们将很乐意为您提供进一步的帮助。
> 
> 十分感谢您的回复，上述问题已解决。请问作者能否分享一下视觉交错CoT数据制作的脚本，这是否适用于不同类型的多模态数学推理数据，供社区调研～～～

Thanks for your interest. We’re organizing the dataset generation scripts and will release them once ready.

**Comment by shuyansy** [comment]
> {'loss': 0.7716, 'grad_norm': 6.769598484039307, 'learning_rate': 9.861932938856016e-08, 'epoch': 0.06} 1%|██▎ | 60/5064 [08:46<11:35:13, 8.34s/it]NaN or Inf found in input tensor. {'loss': 6.8315, 'grad_norm': nan, 'learning_rate': 1.1834319526627217e-07, 'epoch': 0.07} 1%|██▋ | 70/5064 [10:16<12:21:28, 8.91s/it]NaN or Inf found in input tensor. {'loss': 0.0, 'grad_norm': nan, 'learning_rate': 1.3806706114398422e-07, 'epoch': 0.08} 2%|███ | 80/5064 [11:47<12:23:45, 8.95s/it]NaN or Inf found in input tensor. {'loss': 0.0, 'grad_norm': nan, 'learning_rate': 1.5779092702169626e-07, 'epoch': 0.09} 2%|███▏ | 86/5064 [12:44<13:21:40, 9.66s/it] 参数没有改动，请问是哪里有问题呀

@vansuu I met the same problems, could you share how you solved it.

**Comment by HangerYang** [comment]
I met the same problem.

**Comment by ycoliver** [comment]
I met the same problem too, with the experiment conducted on CUDA 12.1, and PyTorch 2.5.1 which the author recommended

{'loss': 1.3619, 'grad_norm': 7.2771806716918945, 'learning_rate': 2.467308166790032e-08, 'epoch': 0.0}
{'loss': 1.3546, 'grad_norm': 69.55654907226562, 'learning_rate': 4.934616333580064e-08, 'epoch': 0.0}
{'loss': 1.5763, 'grad_norm': 9.350918769836426, 'learning_rate': 7.401924500370096e-08, 'epoch': 0.0}
{'loss': 1.4375, 'grad_norm': 10.120987892150879, 'learning_rate': 9.869232667160128e-08, 'epoch': 0.0}
  0%|          | 5/20262 [00:19<22:13:29,  3.95s/it]NaN or Inf found in input tensor.
{'loss': 1.5527, 'grad_norm': nan, 'learning_rate': 1.233654083395016e-07, 'epoch': 0.0}

**Comment by Irisicy4** [comment]
met the same problem; can you please share how you solved it @vansuu
